experiments/pipeline/src/main/scala/org/apache/gearpump/experiments/pipeline/Processors.scala

```diff
@@ -0,0 +1,344 @@
+    }).failed.foreach(LOG.error("bad message", _))
+  }
+
+}
+
+class KafkaProducer(taskContext : TaskContext, conf: UserConfig)
+  extends Task(taskContext, conf) {
+
+  import taskContext.{output, parallelism, taskId}
+
+  private val kafkaConfig = conf.getValue[KafkaConfig](KafkaConfig.NAME).get
+  private val batchSize = kafkaConfig.getConsumerEmitBatchSize
+  private val msgDecoder: MessageDecoder = kafkaConfig.getMessageDecoder
+
+  val taskParallelism = parallelism
+
+  private val source: TimeReplayableSource = new KafkaSource(taskContext.appName, taskId, taskParallelism,
+    kafkaConfig, msgDecoder)
+  private var startTime: TimeStamp = 0L
+
+  override def onStart(newStartTime: StartTime): Unit = {
+    Try({
+      startTime = newStartTime.startTime
+      LOG.info(s"start time is $startTime")
+      source.setStartTime(startTime)
+    }).failed.foreach(LOG.error("caught error", _))
+    self ! Message("start", System.currentTimeMillis())
+  }
+
+  override def onNext(msg: Message): Unit = {
+    Try({
+      source.pull(batchSize).foreach(msg => {
+        output(msg)
+      })
+
+    }).failed.foreach(LOG.error("caught error", _))
+    self ! Message("continue", System.currentTimeMillis())
+  }
+
+  override def onStop(): Unit = {
+    LOG.info("closing kafka source...")
+    source.close()
+  }
+}
+
+trait HBase {
+  this: Task =>
+
+  val ZOOKEEPER = "hbase.zookeeper.connect"
+  val TABLE_NAME = "hbase.table.name"
+  val HBASE_ZOOKEEPER = "hbase.zookeeper.quorum"
+  val hbaseConf = new Configuration
+  val pipelineConfig = taskConfig.getValue[PipelineConfig](PIPELINE)
+  val (zookeepers, tableName) = pipelineConfig.map(config => {
+    val zookeepers = config.getString(ZOOKEEPER)
+    val tableName = config.getString(TABLE_NAME)
+    (zookeepers, tableName)
+  }).get
+  hbaseConf.set(HBASE_ZOOKEEPER, zookeepers)
+  val htd = new HTableDescriptor(TableName.valueOf(tableName))
+  val hcd = new HColumnDescriptor("pipelineFamily")
+  htd.addFamily(hcd)
+  val table = new HTable(hbaseConf, htd.getName)
+
+  /**
+   * Gets the entire contents of the HBase order table and list it to standard output.
+   *
+   * @return the entire order table in ascending rowKey order.
+   */
+  def read:List[Result] = {
+    val scan = new Scan
+    val scanner = table.getScanner(scan)
+    val scalaList: List[Result] = scanner.iterator.toList
+
+    try {
+      for (result <- scalaList) println("Scan 1234: " + result)
+    } finally {
+      scanner.close()
+    }
+    scalaList
+  }
+
+  /**
+   * Returns a HBase result object containing the row and column family fields.
+   *
+   * @return the row pointed to by rowKey
+   */
+  def findByKey(rowKey: String):Result = {
+    val get = new Get(Bytes.toBytes(rowKey))
+    val result = table.get(get)
+    result
+  }
+
+  /**
+   * Deletes all the row pointed to by rowKey including all its column families.
+   */
+  def deleteByKey(rowKey: String) {
```

Comment:
If we want to do this, then definitely in a seperate PR. For this PR, let's keep it simple.